### ISSUE TITLE
Update dependencies

### ISSUE THREAD

**Issue by weberax** [bastislack/highline-freestyle#219]
At some point we should probably update our dependencies... 
Also github reports security vulnerabilities (due to this old dependencies):
> https://github.com/bastislack/highline-freestyle/security/dependabot

Whats the normal workflow for this? 
When to update dependencies?
Should we implement some more unit tests before?

**Comment by weberax** [comment]
@rorystephenson can you help here, with your opinion?

**Comment by rorystephenson** [comment]
Yep a dependency bump is always a good idea. It's worth checking CHANGELOGs if the dependencies have them to be sure that there are no breaking changes and then do some testing before publishing.

Since it's all run local we are not at risk of any security breaches but the more dependencies get behind the more work it is to update them.

**Comment by Dosbodoke** [comment]
I think that updating the dependencies should be done in a safer manner and before it, we need to resolve the error log got from `npm install`

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: react-pwa-install@1.0.12
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.2" from the root project
npm ERR!   peer react@"^16.8.0 || ^17.0.0" from @lingui/react@3.13.2
npm ERR!   node_modules/@lingui/react
npm ERR!     @lingui/react@"^3.13.2" from the root project
npm ERR!     peer @lingui/react@"^3.13.0" from @lingui/macro@3.13.2
npm ERR!     node_modules/@lingui/macro
npm ERR!       dev @lingui/macro@"^3.13.2" from the root project
npm ERR!   32 more (@material-ui/core, @material-ui/styles, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"16.x" from react-pwa-install@1.0.12
npm ERR! node_modules/react-pwa-install
npm ERR!   react-pwa-install@"^1.0.12" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@16.14.0
npm ERR! node_modules/react
npm ERR!   peer react@"16.x" from react-pwa-install@1.0.12
npm ERR!   node_modules/react-pwa-install
npm ERR!     react-pwa-install@"^1.0.12" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/bodok/.npm/_logs/2023-03-17T00_24_21_456Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/bodok/.npm/_logs/2023-03-17T00_24_21_456Z-debug-0.log
```

Based on the error log, the conflicting peer dependency is `react@16.x` required by `react-pwa-install@1.0.12`. However, the current version of React installed is `17.0.2`. Therefore, in order to resolve this conflict,  we need to get rid of  `react-pwa-install`.

This will remove the current PWA implementation (that doesn't work for me)  that needs to be reimplemented and deserves its own issue.

See more on  [**Converting Existing React App to PWA**](https://medium.com/swlh/converting-existing-react-app-to-pwa-3c7e4e773db3)

**Comment by bastislack** [comment]
The problem is that `react-pwa-install` is no longer maintained by the authors. But it is a convenient way to make the install process simple for iOs and Android users.
I will create an issue, so we can try to find an alternative or implement it ourselves.

> This will remove the current PWA implementation (that doesn't work for me) 

What do you mean with it doesn't work for you? The installation? Or general offline usage?

**Comment by Dosbodoke** [comment]
I was trying to install it on Linux and wasn't working, but it should right?

On IOS it worked.

**Comment by JulianDietzel** [comment]
I think this issue should take a high priority, as I am not able to install any npm modules anymore. Trying to do so results in the same error message that @Dosbodoke posted.
